### PR TITLE
Migrate market modal

### LINKED ISSUE #3927: Forking: Migrate Market banner
Migrate Market banner

for markets where reportingState is `AwaitingForkMigration`.

button executes new method in contractsCall.ts (needs to be added) 
`augur/packages/augur-ui/src/modules/modal/containers/modal-migrate-market.ts` modal needs to be finished up. Just need a simple modal.

<https://www.figma.com/file/aAzKHh4cA6OT2t7WFv2BQ7fB/Reporting-and-Disputing?node-id=2146%3A18>

image

design
<https://www.figma.com/file/aAzKHh4cA6OT2t7WFv2BQ7fB/Reporting-and-Disputing?node-id=2114%3A2549>

image

image

### PR DESCRIPTION
**Pull request by adrake33**
Closes #3927. I'm still working on finishing migrateThroughOneFork action, but everything else for this is working, so I'm putting this up now to get the PR process started.

### CODE REVIEW

File: packages/augur-ui/src/modules/forking/actions/migrate-market-through-one-fork.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { closeModal } from 'modules/modal/actions/close-modal';
+import { getGasPrice } from 'modules/auth/selectors/get-gas-price';
+
+export const migrateMarketThroughOneFork = (
+  marketId,
+  estimateGas = false,
+  callback = logError
+) => (dispatch, getState) => {
+  // TBD
+  console.log('in migrateThroughFork');
```
Comment (bthaile) [inline packages/augur-ui/src/modules/forking/actions/migrate-market-through-one-fork.ts]: create call to migrate market in contractCalls.ts

Comment (bthaile) [reply packages/augur-ui/src/modules/forking/actions/migrate-market-through-one-fork.ts]: I see the call in ContractCalls, hook it up here. and add migrate estimate gas method

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 🥳

 This link will always point to this PR's latest changes: https://peek.run/vv2hzxn

**Review by bthaile — CHANGES_REQUESTED** [review]
looks good, wire up contract calls as much as possible.

File: packages/augur-ui/src/modules/market-cards/containers/market-card.ts [hunk 2]
```diff
@@ -34,6 +34,14 @@ const mapDispatchToProps = (dispatch, ownProps) => ({
         selectedOutcome,
       })
     ),
+  migrateMarketModal: () =>
+    dispatch(
+      updateModal({
+        type: MODAL_MIGRATE_MARKET,
+        market: ownProps.market,
+        gasCost: MIGRATE_MARKET_GAS_ESTIMATE,
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/market-cards/containers/market-card.ts]: why are you passing this in here

File: packages/augur-ui/src/modules/reporting/common.tsx [hunk 3]
```diff
@@ -50,14 +60,28 @@ export const DismissableNotice = (props: DismissableNoticeProps) => {
...
+              className={ButtonStyles[props.buttonType]}
+              onClick={props.buttonAction}
+            >
+              {props.buttonText}
+            </button>
+          }
+          {props.buttonType === DISMISSABLE_NOTICE_BUTTON_TYPES.CLOSE &&
+            <button
+              type='button'
+              className={Styles[props.buttonType]}
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/reporting/common.tsx]: i think you can just use buttons here the normal way...?

**Review by phoebemirman — CHANGES_REQUESTED** [review]
few questions. also not sure about button styling thing.

File: packages/augur-ui/src/modules/reporting/common.tsx (hunk 3 above)
Comment (adrake33) [reply packages/augur-ui/src/modules/reporting/common.tsx]: Yeah, I'll just use Styles.PrimaryButton and Styles.close

File: packages/augur-ui/src/modules/market-cards/containers/market-card.ts (hunk 2 above)
Comment (adrake33) [reply packages/augur-ui/src/modules/market-cards/containers/market-card.ts]: Good point, this doesn't need to be passed in. I'll remove it.

**Comment by adrake33** [comment]
If my latest changes look ok, let's get this merged in, and Tom will make the remaining changes to integrate the reporting modal with this so we can get the payout numerators to call migrateThroughOneFork.

**Review by phoebemirman — APPROVED** [review]

**Comment by bthaile** [comment]
I'll work with design to get the reporting form integrated.

**Review by bthaile — APPROVED** [review]
lgtm

**Pull request merged**